packages/react/accordion/src/Accordion.tsx

```diff
@@ -104,10 +101,10 @@ const HEADER_DEFAULT_TAG = 'h3';
  * `AccordionHeader` contains the content for the parts of an `AccordionItem` that will be visible
  * whether or not its content is collapsed.
  */
-const AccordionHeader = forwardRefWithAs<typeof HEADER_DEFAULT_TAG>((props, forwardedRef) => {
+const AccordionHeader = React.forwardRef((props, forwardedRef) => {
```

Comment:
Per my comment in the main thread, would we not need type annotations on the inner function params here internally?

Reply:
I've just checked out the PR and as far as I can tell all the inner typing is working correctly. You get full types for props and forwardedRef.

Reply:
Yeah, this work because of this sneaky little bit at the bottom 😛 

https://github.com/radix-ui/primitives/pull/396/files#diff-fe43cc8774290b4daed95c5a7b644d0f1e4d25984d70e152499a592613884b7bR71